### DISCUSSION TITLE
Too many open files - inside Docker?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#3703] [Q&A]
We are seeing errors like the one below happening inside the Docker container for the instance. Would this indicate a host-level failure, or something they're doing inside the container?

```
[2023-11-07T15:57:33.745Z] [WARN] panic: registerHooks: too many open files
[2023-11-07T15:57:33.746Z] [WARN] goroutine 1 [running]:
[2023-11-07T15:57:33.746Z] [WARN] github.com/pocketbase/pocketbase/plugins/jsvm.MustRegister(...)
[2023-11-07T15:57:33.746Z] [WARN]     /home/runner/work/pocketbase/pocketbase/plugins/jsvm/jsvm.go:91
[2023-11-07T15:57:33.746Z] [WARN] main.main()
[2023-11-07T15:57:33.746Z] [WARN]     /home/runner/work/pocketbase/pocketbase/examples/base/main.go:96 +0x66e
```

**Comment by ganigeorgiev** [comment]
Hm, I'm not sure whether it is caused by a host-level failure or from something in the container.

Usually this error is returned when the file descriptors limit for the user executing the PocketBase process has been reached (see https://pocketbase.io/docs/going-to-production/#increase-the-open-file-descriptors-limit).

It can be changed via `ulimit` or if you use systemd service(s) via the `LimitNOFILE` option.
If you use `docker run` there is also a [`--ulimit` flag](https://docs.docker.com/engine/reference/commandline/run/).

It is also possible to be some obscure bug with PocketBase (eg. forgotten `Close()` call) but at least locally so far I haven't stumbled on it (including when running the [benchmarks](https://github.com/pocketbase/benchmarks)).

**Reply by benallfree** [reply]
I did a little research and it appears that the host ulimit affects all containers combined. We've been running 40+ containers simultaneously, it's possible they were hitting the host's 1024 ulimit with all of the combined activity. I adjusted the host ulimit and imposed a ulimit of 1024 per container, so we'll see if this alleviates the problem.
